src/Fixer/ControlStructure/NoBreakCommentFixer.php

```diff
@@ -117,22 +117,29 @@ static function ($value) {
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        for ($position = \count($tokens) - 1; $position >= 0; --$position) {
-            if ($tokens[$position]->isGivenKind([T_CASE, T_DEFAULT])) {
-                $this->fixCase($tokens, $position);
+        for ($index = \count($tokens) - 1; $index >= 0; --$index) {
+            if (!$tokens[$index]->isGivenKind([T_CASE, T_DEFAULT])) {
+                continue;
             }
+
+            $caseColonIndex = $tokens->getNextTokenOfKind($index, [':', ';', [T_DOUBLE_ARROW]]);
+
+            if ($tokens[$index]->isGivenKind(T_DOUBLE_ARROW)) {
```

Comment:
Did you mean
```diff
-if ($tokens[$index]->isGivenKind(T_DOUBLE_ARROW)) {
+if ($tokens[$caseColonIndex]->isGivenKind(T_DOUBLE_ARROW)) {
```
?

Currently I think the statement is always `false`. The test pass though, so I'm kind of confused 🤔

Reply:
Oh, I get it - in the `fixCase` there is a check for another case/default section, and with only "default" in "match" there is none, so the if check is here only to save time.

The bug was that `$tokens->getNextTokenOfKind($index, [':', ';', [T_DOUBLE_ARROW]]);` when it was without `T_DOUBLE_ARROW` if was finding `;` far away (not as the very next meaningful token) and the `default` was treated as from other case/default section.